js_sdk/end_idle_engagements.js

```diff
@@ -0,0 +1,32 @@
+// Reference:
+// https://js-sdk-docs.salemove.com/class/Engagement.html
+// Running the code snippet below before starting an engagement
+// allows to track Visitor activity to end the ongoing engagement
+// when Visitor has been idle (no mouse or keyboard event recorded)
+// for some given amount of time
```

Comment:
Is this a reliable heuristic that we are confident enough to suggest to our clients? If so, why isn't it a customizable part of the platform itself?

Reply:
The initial request came from the client Belfius as they wanted to kill ongoing "idle" engagements using JS SDK. I'm not sure about the confidence level but suggesting this to our clients under the MIT License seems good enough for me. In order to integrate similar functionality to our platform, it would need first some additional research. PS. This code snippet won't work 100% when engagement_mode = 'iframed' but we don't have currently many production sites with this configuration. However, this support can perhaps be added to this example later.

Reply:
> I'm not sure about the confidence level but suggesting this to our clients under the MIT License seems good enough for me. In order to integrate similar functionality to our platform, it would need first some additional research.

Why does it require further research for integrating into the platform but not for suggesting to the client?

Reply:
Because when this is a part of our platform then we are fully liable for the code. In that case, further research needs to be done in my opinion.

Reply:
Aren't we just as liable with these suggestions? Client has a problem with idle visitors. What's the difference between us telling them 1) use this code snippet vs 2) enable this toggle for your site?

Reply:
In both cases if it starts causing problems then we're the ones responsible, no?

Reply:
If you think liability is a problem here then we can remove the exact steps for tracking visitor activity and let the client figure it out on their own. In that case, we would be more sharing some ideas on how to do this rather than giving a copy-paste solution. WDYT?

For custom code examples, we are asking clients to sign a disclaimer that the custom code is not guaranteed by Glia and is being provided only as an example. Perhaps the same should apply here as well.

Reply:
I guess I just don't really like that solution here. What if the visitor is talking with the operator on the phone and they don't need to touch the keyboard or mouse for a few minutes? Imagine the SSDs you might start getting when they implement something like this and then it starts behaving surprisingly. It could be a big time sink for CEs trying track down why "engagements are dropping in the middle of conversation" or similar

Reply:
If you send this as "here's something you could possibly do, but it's not really reliable in all cases and we will not give any guarantees on this",  I guess I would be OK with this. But this is really more of a call for your own team. Maybe talk this through with @p2tris.

Reply:
@p2tris, please share your thoughts on this. Shall we keep or remove this example? I mean, it was never meant to be a public example for the clients but as it was suggested to put it to Github (to be reused later when needed) then this repo seemed to be relevant for me.

Reply:
I think this example should be here (useful example of JS SDK use). Deiwin also has a good point of it being as a feature flag in site settings - perhaps you can discuss it with Andreas?

Reply:
@andreassisask, WDYT? Could we consider adding visitor idle time tracking as a customizable part of the Glia platform? By the idle time, I mean no mouse or keyboard event recorded for some configurable amount of time.